libraries/AP_BattMonitor/AP_BattMonitor_DJI.cpp

```diff
@@ -0,0 +1,79 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_BattMonitor_DJI.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_Math/crc.h>
+
+/*
+  "battery" monitor for DJI Battery via serial port.
+
+   For future Referance:
+   Total Voltage : 8,9
+   Percentage : 17
+   Cell1 :  20,21
+   Cell2 :  22,23
+   Cell3 :  24,25
+   Cell4 :  26,27
+   
+ */
+extern const AP_HAL::HAL& hal;
+
+ #define DELAY_TIME_US      1000000     // Battery info requesting delay
+
+/// Constructor
+AP_BattMonitor_DJI::AP_BattMonitor_DJI(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    
+    if ((port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_DJIBattery, 0))) {
+        last_send_us = AP_HAL::micros();
+    } else { 
+        gcs().send_text(MAV_SEVERITY_WARNING,"DJIBattery: Port not available or not configured properly");
+    }
+    // need to add check
+    _state.healthy = false;
+}
+
+/*
+  read - read the "voltage" and "percentage"
+*/
+void AP_BattMonitor_DJI::read()
+{
+    if (port != nullptr){
+        uint32_t now = AP_HAL::micros();
+        if (last_send_us != 0 && now - last_send_us > DELAY_TIME_US) {
+           
+            for (uint8_t i=0; i < count; i++) {
```

Comment:
You don't need to clear the buffer if you are resetting the count, because you won't use the data later unless you copied into it.

Reply:
yes, will remove the loop